tests/common/global_psp_checks.go

```diff
@@ -0,0 +1,50 @@
+package common
+
+import (
+	"github.com/rancher/hull/pkg/checker"
+	"github.com/rancher/hull/pkg/test"
+	"github.com/stretchr/testify/assert"
+	policyv1 "k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/kubernetes/pkg/apis/rbac"
+)
+
+func GetPspEnabledCheck(numPspsWithPspEnabled int) test.Checks {
+	return test.Checks{
+		checker.OnResources(func(tc *checker.TestContext, pspList []*policyv1.PodSecurityPolicy) {
+
+			pspsEnabled, _ := checker.RenderValue[bool](tc, ".Values.global.cattle.psp.enabled")
+
+			if pspsEnabled {
+				assert.Equal(tc.T, numPspsWithPspEnabled, len(pspList), "total number of psps mismatch, expected:%d, got: %d",
+					numPspsWithPspEnabled, len(pspList))
+			}
+
+		}),
+		checker.PerResource(func(tc *checker.TestContext, clusterRole *rbacv1.ClusterRole) {
+			pspsEnabled, _ := checker.RenderValue[bool](tc, ".Values.global.cattle.psp.enabled")
+			if !pspsEnabled {
+				for _, rule := range clusterRole.Rules {
+					for _, resource := range rule.Resources {
+						if resource == "podsecuritypolicies" {
+							tc.T.Errorf("err: ClusterRole %s has reference to psp resources with psp disabled", clusterRole.Name)
+							return
+						}
+					}
+				}
+			}
+		}),
+		checker.PerResource(func(tc *checker.TestContext, role *rbacv1.Role) {
+			pspsEnabled, _ := checker.RenderValue[bool](tc, ".Values.global.cattle.psp.enabled")
+			if !pspsEnabled {
+				for _, rule := range role.Rules {
+					for _, resource := range rule.Resources {
+						if resource == "podsecuritypolicies" {
+							tc.T.Errorf("err: Role %s has reference to psp resources with psp disabled", role.Name)
+							return
+						}
+					}
```

Comment:
To avoid duplicating logic, perhaps it might make sense to introduce an RBAC-related looper like we do for workloads here: https://github.com/rancher/hull/blob/824596116fec7cf909442e4186d210fe9022a1e3/pkg/checker/loop.go#L37-L70

cc: @eliyamlevy